src/services/ghost/classic-auto-complete/GhostInlineCompletionProvider.ts

```diff
@@ -280,15 +271,26 @@ export class GhostInlineCompletionProvider implements vscode.InlineCompletionIte
 		}
 
 		// No cached suggestion available - invoke LLM
-		if (this.model && this.ghostContext) {
-			const context: GhostSuggestionContext = {
-				document,
-				range: new vscode.Range(position, position),
+		if (this.model) {
+			// Build autocomplete input directly with all tracking data
+			const recentlyVisitedRanges = this.recentlyVisitedRangesService.getSnippets()
+			const recentlyEditedRanges = await this.recentlyEditedTracker.getRecentlyEditedRanges()
+
+			const autocompleteInput: AutocompleteInput = {
```

Comment:
building the autocomplete input here is a regression imho; it is the input for the completion provider. The reason to keep it outside is that it makes it easier to be more compatible with continue. If we don't care about that, there is no reason to use this type at all